lib/rules/no-lone-blocks.js

```diff
@@ -11,12 +11,46 @@
 
 module.exports = function(context) {
 
+    /**
+     * Checks if the block includes any block-level bindings
+     * @param {ASTNode} block - The block node to be checked.
+     * @param {boolean} isStrict - True if the block scope is in strict mode.
+     * @returns {boolean} True if the block includes block-level bindings.
+    */
+    function hasBlockBindings(block, isStrict) {
+        var statement,
+            i, l;
+
+        for (i = 0, l = block.body.length; i < l; ++i) {
```

Comment:
This is expensive - you're doing another traversal even though ESLint is already doing one.

I'm actually not sure this is even necessary. Why not just don't warn when blockBindings is true?